internal/cmd/logout.go

```diff
@@ -36,20 +81,94 @@ func logout(purge bool) error {
 		return err
 	}
 
-	for i, hostConfig := range config.Hosts {
-		config.Hosts[i].AccessKey = ""
-
-		client, err := apiClient(hostConfig.Host, hostConfig.AccessKey, hostConfig.SkipTLSVerify)
-		if err != nil {
-			logging.S.Warn(err.Error())
-			continue
+	switch {
```

Comment:
This is hard to understand. It would be easier to read and understand if some of the shared functionality is split out:

```go
func logout(server, all, clear) error {
  config, err := readConfig()
  if err != nil {
    return err
  }

  for i := range config.Servers {
    if all || config.Servers[i].Server == server || config.Servers.[i].Current {
      if err := logoutServer(&config.Servers[i], clear); err != nil {
        return err
      }
    }

    if config.Servers[i] == nil {
      config.Servers = append(config.Servers[:i], config.Servers[i+1:]...)
    }
  }

  if err := writeConfig(config); err != nil {
    return err
  }

  return nil
}

func logoutServer(config *ClientHostConfig, clear bool) error {
  client, err := apiClient(config.Server, config.AccessKey, config.SkipTLSVerify)
  if err != nil {
    return err
  }

  _ = client.Logout()

  if clear {
    client = nil
    return nil
  }

  client.AccessKey = ""
  return nil
}
```

Reply:
done